build-tools-internal/src/test/java/org/elasticsearch/gradle/internal/release/AddVersionTaskTests.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.release;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter;
+
+import org.elasticsearch.gradle.Version;
+import org.junit.Test;
+
+import java.io.StringWriter;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertFalse;
+
+public class AddVersionTaskTests {
+
+    @Test
+    public void addVersion_versionExists() {
+        final String versionJava = """
+            public class Version {
+                public static final Version V_8_10_0 = new Version(8_10_00_99);
+                public static final Version V_8_10_1 = new Version(8_10_01_99);
+                public static final Version V_8_11_0 = new Version(8_11_00_99);
+                public static final Version CURRENT = V_8_11_0;
+            }""";
+
+        CompilationUnit unit = StaticJavaParser.parse(versionJava);
+
+        var newUnit = AddVersionTask.addVersionConstant(unit, Version.fromString("8.10.1"), false);
+        assertThat(newUnit.isPresent(), is(false));
+    }
+
+    @Test
+    public void addVersion_oldVersion() {
+        final String versionJava = """
+            public class Version {
+                public static final Version V_8_10_0 = new Version(8_10_00_99);
+                public static final Version V_8_10_1 = new Version(8_10_01_99);
+                public static final Version V_8_11_0 = new Version(8_11_00_99);
+                public static final Version CURRENT = V_8_11_0;
+            }""";
+        final String updatedVersionJava = """
+            public class Version {
```

Comment:
I notice the formatting is different here. I presume that is because we are relying on the `toString()` implementation instead of using a lexical output implementation. Should we be doing that so our tests better mimic what the actual task implementation is doing?

Reply:
This is the default formatting output of javaparser. I didn't want to complicate it with lexical information, to get a 'pure' test of adding the version to the AST. Reading/writing the actual file are tested below